app/models/ext_management_system.rb

```diff
@@ -634,7 +634,7 @@ def build_endpoint_by_role(options)
     endpoint = endpoints.detect { |e| e.role == options[:role].to_s }
     # update or create
     if endpoint
-      endpoint.update(options)
+      endpoint.assign_attributes(options)
```

Comment:
Makes sense... is this ever called where the caller would expect the endpoint to be saved?

Code comments.... ☝️  Please update the comment on 635 to indicate we're setting the attributes in an endpoint or building a new one without saving... At the very least, "without saving should be mentioned as to why we're not doing update/create.